README.md
# Skada for WotLK `3.3.5` (_Revisited - v1.8.84_)

![Discord](https://img.shields.io/discord/795698054371868743?label=discord)
![GitHub last commit](https://img.shields.io/github/last-commit/bkader/Skada-WoTLK)
![GitHub tag (latest by date)](https://img.shields.io/github/v/tag/bkader/Skada-WoTLK?label=version)

Skada is a modular damage meter with various viewing modes, segmented fights and customizable windows. It aims to be highly efficient with memory and CPU.

<p align="center"><img src="https://user-images.githubusercontent.com/4732702/170839578-72a9a952-c999-457a-8f57-7d151e3b76a8.png" alt="Skada WotLK"></p>

_**FOR PRIVATE SERVERS**_:
* _[Cataclysm](https://github.com/bkader/Skada-Cata)_
* _[Mists of Pandaria](https://github.com/bkader/Skada-MoP)_

_**FOR OFFICIAL SERVERS**_:
* _[Retail/Classic Wrath](https://github.com/bkader/Skada-Damage-Meter)_

> ***Is this a backported version?***
> **No**! This is total **rewrite** of the original addon, it all started from version `r301` for `3.3.5`and built up to the current state.

## IMPORTANT: How to install

1. If you used the default on **Skada** before, please make sure to delete all its files from `Interface\AddOns` folder as well as all its _SavedVariables_ from `WTF` folder (_just delete all `Skada.lua` and `Skada.lua.bak` for this folder. Use the search box for quick delete_). If you are new, skip this step.
2. [Download the package](https://github.com/bkader/Skada-WoTLK/archive/refs/heads/main.zip).
3. Open the Zip package inside which you will find a single folder named `Skada-WoTLK-main`.
4. Extract or drag and drop the unique folder `Skada` into your `Interface\AddOns` folder.
5. If you want to use `SkadaImprovement` and/or `SkadaStorage` modules, drop them there as well.

## Show Love & Support

Though it's not required, **PayPal**/**Paysera** donations are most welcome at **bkader[at]mail.com**.

## Special Credits

* Original author: **Zarnivoop**
* Localization: **meatgaga** (CN), **Ícar** & **Septimun** (ES), **Kader** (FR), **NGL** (RU)
* Direct and indirect contributors: **Abel**, **Iqui**, **Jeb**, **Shoggoth**, **Havi** & **Ganrod**, **WotLK Community**.
* The man behind the wheel: **[Nomadra](https://github.com/ridepad)**

## OTHER ADDONS (Private Servers)

* **!ElvinCDs**: [https://github.com/bkader/ElvinCDs](https://github.com/bkader/ElvinCDs)
* **!KTracker**: [https://github.com/bkader/KTracker](https://github.com/bkader/KTracker)
* **BigWigs**: [https://github.com/bkader/BigWigs-WoTLK](https://github.com/bkader/BigWigs-WoTLK)
* **BloodyMask**: [https://github.com/bkader/BloodyMask](https://github.com/bkader/BloodyMask)
* **ButtonFacade**: [https://github.com/bkader/ButtonFacade](https://github.com/bkader/ButtonFacade)
* **Comix**: [https://github.com/bkader/Comix-WoTLK](https://github.com/bkader/Comix-WoTLK)
* **Crosshairs**: [https://github.com/bkader/Crosshairs-WotLK](https://github.com/bkader/Crosshairs-WotLK)
* **Dominos**: [https://github.com/bkader/Dominos](https://github.com/bkader/Dominos)
* **EclipseBar**: [https://github.com/bkader/EclipseBar-WotLK](https://github.com/bkader/EclipseBar-WotLK)
* **FixGroups**: [https://github.com/bkader/FixGroups-WoTLK](https://github.com/bkader/FixGroups-WoTLK)
* **GarbageProtector**: [https://github.com/bkader/GarbageProtector](https://github.com/bkader/GarbageProtector)
* **Grid2**: [https://github.com/bkader/Grid2-WoTLK](https://github.com/bkader/Grid2-WoTLK)
* **HalionHelper**: [https://github.com/bkader/HalionHelper-WoTLK](https://github.com/bkader/HalionHelper-WoTLK)
* **Hermes**: [https://github.com/bkader/Hermes-WotLK](https://github.com/bkader/Hermes-WotLK)
* **KPack**: [https://github.com/bkader/KPack](https://github.com/bkader/KPack)
* **KRaidTools**: [https://github.com/bkader/KRT](https://github.com/bkader/KRT)
* **KRaidUtilities**: [https://github.com/bkader/KRU-WoTLK](https://github.com/bkader/KRU-WoTLK)
* **KuiNameplates**: [https://github.com/bkader/KuiNameplates-WoTLK](https://github.com/bkader/KuiNameplates-WoTLK)
* **Masque**: [https://github.com/bkader/Masque-WoTLK](https://github.com/bkader/Masque-WoTLK)
* **NameplateSCT**: [https://github.com/bkader/NameplateSCT_WoTLK](https://github.com/bkader/NameplateSCT_WoTLK)
* **PlateBuffs**: [https://github.com/bkader/PlateBuffs_WoTLK](https://github.com/bkader/PlateBuffs_WoTLK)
* **RUF**: [https://github.com/bkader/RUF-WoTLK](https://github.com/bkader/RUF-WoTLK)
* **Raven**: [https://github.com/bkader/Raven-WotLK](https://github.com/bkader/Raven-WotLK)
* **SharedMedia**: [https://github.com/bkader/SharedMedia](https://github.com/bkader/SharedMedia)
* **Talented**: [https://github.com/bkader/Talented_WoTLK](https://github.com/bkader/Talented_WoTLK)
* **TellMeWhen**: [https://github.com/bkader/TellMeWhen_3.3.5](https://github.com/bkader/TellMeWhen_3.3.5)
* **TheClassicRace**: [https://github.com/bkader/TheClassicRace-WotLK](https://github.com/bkader/TheClassicRace-WotLK)
* **TidyPlates & ThreatPlates**: [https://github.com/bkader/TidyPlates_WoTLK](https://github.com/bkader/TidyPlates_WoTLK)

CHANGELOG.md
# Changelog

## [1.8.84] - 2022-11-25

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.83...1.8.84)

## [1.8.83] - 2022-11-02

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.82...1.8.83)

## [1.8.82] - 2022-10-12

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.81...1.8.82)

## [1.8.81] - 2022-09-21

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.80...1.8.81)

## [1.8.80] - 2022-09-13

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.79...1.8.80)

## [1.8.79] - 2022-06-30

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.78...1.8.79)

## [1.8.78] - 2022-06-12

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.77...1.8.78)

## [1.8.77] - 2022-05-12

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.76...1.8.77)

## [1.8.76] - 2022-04-14

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.75...1.8.76)

## [1.8.75] - 2022-04-05

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.74...1.8.75)

## [1.8.74] - 2022-03-27

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.362...1.8.74)

## [1.8.73.362] - 2022-03-09

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.361...1.8.73.362)

## [1.8.73.361] - 2022-03-03

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.360...1.8.73.361)

## [1.8.73.360] - 2022-02-18

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.359...1.8.73.360)

## [1.8.73.359] - 2022-02-10

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.358...1.8.73.359)

## [1.8.73.358] - 2022-02-09

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.357...1.8.73.358)

## [1.8.73.357] - 2022-02-09

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.356...1.8.73.357)

## [1.8.73.356] - 2021-12-29

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.355...1.8.73.356)

## [1.8.73.355] - 2021-12-29

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.354...1.8.73.355)

## [1.8.73.354] - 2021-12-28

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.353...1.8.73.354)

## [1.8.73.353] - 2021-12-27

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.352...1.8.73.353)

## [1.8.73.352] - 2021-12-20

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.351...1.8.73.352)

## [1.8.73.351] - 2021-12-20

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.350...1.8.73.351)

## [1.8.73.350] - 2021-12-20

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.349...1.8.73.350)

## [1.8.73.349] - 2021-12-19

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.348...1.8.73.349)

## [1.8.73.348] - 2021-12-17

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.347...1.8.73.348)

## [1.8.73.347] - 2021-12-14

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.346...1.8.73.347)

## [1.8.73.346] - 2021-12-14

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.345...1.8.73.346)

## [1.8.73.345] - 2021-12-13

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.344...1.8.73.345)

## [1.8.73.344] - 2021-12-13

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.343...1.8.73.344)

## [1.8.73.343] - 2021-12-11

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.340...1.8.73.343)

## [1.8.73.340] - 2021-12-11

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.338...1.8.73.340)

## [1.8.73.338] - 2021-12-11

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.334...1.8.73.338)

## [1.8.73.334] - 2021-11-30

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.333...1.8.73.334)

## [1.8.73.333] - 2021-11-30

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.332...1.8.73.333)

## [1.8.73.332] - 2021-11-29

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.330...1.8.73.332)

## [1.8.73.330] - 2021-11-26

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73.329...1.8.73.330)

## [1.8.73.329] - 2021-11-26

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.73...1.8.73.329)

## [1.8.73] - 2021-11-21

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.72.21321...1.8.73)

## [1.8.72.21321] - 2021-11-17

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.72...1.8.72.21321)

## [1.8.72] - 2021-11-16

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.71...1.8.72)

## [1.8.71] - 2021-10-26

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.70...1.8.71)

## [1.8.70] - 2021-10-03

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.69...1.8.70)

## [1.8.69] - 2021-09-16

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.68...1.8.69)

## [1.8.68] - 2021-09-04

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.67-R7...1.8.68)

## [1.8.67-rev7] - 2021-08-30

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.67-R6...1.8.67-R7)

## [1.8.67-rev6] - 2021-08-30

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.67-R5...1.8.67-R6)

## [1.8.67-rev5] - 2021-08-29

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.67-R4...1.8.67-R5)

## [1.8.67-rev4] - 2021-08-28

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.67-R3...1.8.67-R4)

## [1.8.67-rev3] - 2021-08-27

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.67...1.8.67-R3)

## [1.8.67] - 2021-08-26

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.66-rev8...1.8.67)

## [1.8.66-rev8] - 2021-08-23

[view diff](https://github.com/bkader/Skada-WoTLK/compare/1.8.42...1.8.66-rev8)

Options.lua
local Skada=Skada
if not Skada then return end

local L=LibStub("AceLocale-3.0"):GetLocale("Skada", false)

Skada.resetoptions={[1]=NO, [2]=YES, [3]=L["Ask"]}

Skada.windowdefaults={
  name="Skada",

  barspacing=0,
  bartexture="BantoBar",
  barfont="Accidental Presidency",
  barfontflags="",
  barfontsize=13,
  numfont="Accidental Presidency",
  numfontflags="",
  numfontsize=13,
  barheight=18,
  barwidth=240,
  barorientation=1,
  barcolor={r=0.3, g=0.3, b=0.8, a=1},
  barbgcolor={r=0.3, g=0.3, b=0.3, a=0.6},
  barslocked=false,
  clickthrough=false,

  spellschoolcolors=true,
  classcolorbars=true,
  classcolortext=false,
  classicons=true,
  roleicons=false,
  showself=true,

  -- buttons
  buttons={menu=true, reset=true, report=true, mode=true, segment=true},
  
  -- title options
  title={
    height=20,
    font="Accidental Presidency",
    fontsize=13,
    fontflags="",
    color={r=0.3,g=0.3,b=0.3,a=1},
    texture="Armory",
    textcolor={r=0.9, g=0.9, b=0.9, a=1},
    bordertexture="None",
    borderthickness=2,
    bordercolor={r=0,g=0,b=0,a=1},
  },

  background={
    tile=false, 
    tilesize=0, 
    color={r=0,g=0,b=0,a=0.4}, 
    texture="Solid",
    bordercolor={r=0,g=0,b=0,a=0.5},
    bordertexture="Blizzard Party", 
    borderthickness=1,
    height=200,
  },

  strata="LOW",
  scale=1,


  reversegrowth=false,
  modeincombat="",
  returnaftercombat=false,
  wipemode="",
  
  hidden=false,
  enabletitle=true,
  titleset=true,
  
  set="current",
  mode=nil,
  
  display="bar",
  snapto=true,
  smoothing=false,

  -- inline bar display
  isonnewline=false,
  isusingclasscolors=true,
  height=30,
  width=600,
  color={r=0.3, g=0.3, b=0.3, a=0.6},
  isusingelvuiskin=true,
  issolidbackdrop=false,
  fixedbarwidth=false,
}

local windefaultscopy={}
Skada:tcopy(windefaultscopy, Skada.windowdefaults)

Skada.defaults={
  profile={
    reset={instance=1, join=3, leave=1},
    icon={hide=false, radius=80, minimapPos=195},
    numberformat=1,
    setformat=3,
    setnumber=true,
    showranks=true,
    setstokeep=10,
    tooltips=true,
    tooltippos="smart",
    tooltiprows=3,
    informativetooltips=true,
    onlykeepbosses=false,
    tentativecombatstart=false,
    hidesolo=false,
    hidepvp=false,
    hidedisables=true,
    hidecombat=false,
    mergepets=true,
    feed="",
    showtotals=false,
    autostop=false,
    sortmodesbyusage=false,
    updatefrequency=0.25,

    modules={},
    columns={},
    report={mode="Damage", set="current", channel="Say", chantype="preset", number=10},
    modulesBlocked={},

    windows={windefaultscopy},
  }
}

-- Adds column configuration options for a mode.
function Skada:AddColumnOptions(mod)
  local db=self.db.profile.columns

  if mod.metadata and mod.metadata.columns then
    local cols={
      type="group",
      name=mod:GetName(),
      order=0,
      inline=true,
      args={}
    }

    for colname, value in pairs(mod.metadata.columns) do
      local c=mod:GetName().."_"..colname

      -- Set initial value from db if available, otherwise use mod default value.
      if db[c]~=nil then
        mod.metadata.columns[colname]=db[c]
      end

      -- Add column option.
      local col={
        type="toggle",
        name=L[colname] or colname,
        get=function() return mod.metadata.columns[colname] end,
        set=function()
        mod.metadata.columns[colname]=not mod.metadata.columns[colname]
        db[c]=mod.metadata.columns[colname]
        Skada:UpdateDisplay(true)
        end,
      }
      cols.args[c]=col
    end

    Skada.options.args.columns.args[mod:GetName()]=cols
  end

end

do
  local numorder=1
  function Skada:AddLoadableModuleCheckbox(mod, name, description)
    local new={
      type="toggle",
      name=name,
      desc=description,
      order=numorder
    }
    Skada.options.args.modules.args[mod]=new
    numorder=numorder+1
  end
end

local deletewindow=nil

Skada.options={
  type="group",
  name="Skada",
  plugins={},
  args={
    d={
      type="description",
      name=L["A damage meter."],
      order=0,
    },

    windows={
      type="group",
      name=L["Windows"],
      order=0,
      args={

        create={
          type="input",
          name=L["Create window"],
          desc=L["Enter the name for the new window."],
          set=function(self, val) if val and val~="" then Skada:CreateWindow(val) end end,
          order=1,
        },

        delete={
          type="select",
          name=L["Delete window"],
          desc=L["Choose the window to be deleted."],
          values= function()
          local windows={}
          for i, win in ipairs(Skada:GetWindows()) do
            windows[win.db.name]=win.db.name
          end
          return windows
          end,
          get=function() return deletewindow end,
          set=function(self, val) deletewindow=val end,
          order=2,
        },
        deleteexecute={
          type="execute",
          name=L["Delete window"],
          desc=L["Deletes the chosen window."],
          func=function(self) if deletewindow then Skada:DeleteWindow(deletewindow) end end,
          order=3,
        },

      },
    },

    resetoptions={
      type="group",
      name=L["Data resets"],
      order=2,
      args={

        resetinstance={
          type="select",
          name=L["Reset on entering instance"],
          desc=L["Controls if data is reset when you enter an instance."],
          values= function() return Skada.resetoptions end,
          get=function() return Skada.db.profile.reset.instance end,
          set=function(self, opt) Skada.db.profile.reset.instance=opt end,
          order=30,
        },

        resetjoin={
          type="select",
          name=L["Reset on joining a group"],
          desc=L["Controls if data is reset when you join a group."],
          values= function() return Skada.resetoptions end,
          get=function() return Skada.db.profile.reset.join end,
          set=function(self, opt) Skada.db.profile.reset.join=opt end,
          order=31,
        },

        resetleave={
          type="select",
          name=L["Reset on leaving a group"],
          desc=L["Controls if data is reset when you leave a group."],
          values= function() return Skada.resetoptions end,
          get=function() return Skada.db.profile.reset.leave end,
          set=function(self, opt) Skada.db.profile.reset.leave=opt end,
          order=32,
        },

      }

    },

    generaloptions={
      type="group",
      name=L["General options"],
      order=3,
      args={

        mmbutton={
          type="toggle",
          name=L["Show minimap button"],
          desc=L["Toggles showing the minimap button."],
          order=1,
          get=function() return not Skada.db.profile.icon.hide end,
          set=function()
          Skada.db.profile.icon.hide=not Skada.db.profile.icon.hide
          Skada:RefreshMMButton()
          end
        },

        mergepets={
          type="toggle",
          name=L["Merge pets"],
          desc=L["Merges pets with their owners. Changing this only affects new data."],
          order=2,
          get=function() return Skada.db.profile.mergepets end,
          set=function() Skada.db.profile.mergepets=not Skada.db.profile.mergepets end,
        },

        showtotals={
          type="toggle",
          name=L["Show totals"],
          desc=L["Shows a extra row with a summary in certain modes."],
          order=3,
          get=function() return Skada.db.profile.showtotals end,
          set=function() Skada.db.profile.showtotals=not Skada.db.profile.showtotals end,
        },

        onlykeepbosses={
          type="toggle",
          name=L["Only keep boss fighs"],
          desc=L["Boss fights will be kept with this on, and non-boss fights are discarded."],
          order=4,
          get=function() return Skada.db.profile.onlykeepbosses end,
          set=function() Skada.db.profile.onlykeepbosses=not Skada.db.profile.onlykeepbosses end,
        },

        hidesolo={
          type="toggle",
          name=L["Hide when solo"],
          desc=L["Hides Skada's window when not in a party or raid."],
          order=5,
          get=function() return Skada.db.profile.hidesolo end,
          set=function()
            Skada.db.profile.hidesolo=not Skada.db.profile.hidesolo
            Skada:ApplySettings()
          end
        },

        hidepvp={
          type="toggle",
          name=L["Hide in PvP"],
          desc=L["Hides Skada's window when in Battlegrounds/Arenas."],
          order=6,
          get=function() return Skada.db.profile.hidepvp end,
          set=function()
            Skada.db.profile.hidepvp=not Skada.db.profile.hidepvp
            Skada:ApplySettings()
          end
        },

        hidecombat={
          type="toggle",
          name=L["Hide in combat"],
          desc=L["Hides Skada's window when in combat."],
          order=7,
          get=function() return Skada.db.profile.hidecombat end,
          set=function()
            Skada.db.profile.hidecombat=not Skada.db.profile.hidecombat
            Skada:ApplySettings()
          end
        },

        disablewhenhidden={
          type="toggle",
          name=L["Disable while hidden"],
          desc=L["Skada will not collect any data when automatically hidden."],
          order=8,
          get=function() return Skada.db.profile.hidedisables end,
          set=function()
          Skada.db.profile.hidedisables=not Skada.db.profile.hidedisables
          Skada:ApplySettings()
          end
        },

        sortmodesbyusage={
          type="toggle",
          name=L["Sort modes by usage"],
          desc=L["The mode list will be sorted to reflect usage instead of alphabetically."],
          order=9,
          width="full",
          get=function() return Skada.db.profile.sortmodesbyusage end,
          set=function()
            Skada.db.profile.sortmodesbyusage=not Skada.db.profile.sortmodesbyusage
            Skada:ApplySettings()
          end
        },

        showranks={
          type="toggle",
          name=L["Show rank numbers"],
          desc=L["Shows numbers for relative ranks for modes where it is applicable."],
          order=10,
          get=function() return Skada.db.profile.showranks end,
          set=function()
          Skada.db.profile.showranks=not Skada.db.profile.showranks
          Skada:ApplySettings()
          end
        },

        tentativecombatstart={
          type="toggle",
          name=L["Aggressive combat detection"],
          desc=L["Skada usually uses a very conservative (simple) combat detection scheme that works best in raids. With this option Skada attempts to emulate other damage meters. Useful for running dungeons. Meaningless on boss encounters."],
          order=11,
          get=function() return Skada.db.profile.tentativecombatstart end,
          set=function() Skada.db.profile.tentativecombatstart=not Skada.db.profile.tentativecombatstart end,
        },

        autostop={
          type="toggle",
          name=L["Autostop"],
          desc=L["Automatically stops the current segment after half of all raid members have died."],
          order=12,
          get=function() return Skada.db.profile.autostop end,
          set=function() Skada.db.profile.autostop=not Skada.db.profile.autostop end
        },

        showself={
          type="toggle",
          name=L["Always show self"],
          desc=L["Keeps the player shown last even if there is not enough space."],
          order=13,
          get=function() return Skada.db.profile.showself end,
          set=function() 
            Skada.db.profile.showself=not Skada.db.profile.showself
            Skada:ApplySettings()
          end
        },

        numberformat={
          type="select",
          name=L["Number format"],
          desc=L["Controls the way large numbers are displayed."],
          order=14,
          values= function() return {[1]=L["Condensed"], [2]=L["Detailed"]} end,
          get=function() return Skada.db.profile.numberformat end,
          set=function(self, opt) Skada.db.profile.numberformat=opt end,
        },

        datafeed={
          type="select",
          name=L["Data feed"],
          desc=L["Choose which data feed to show in the DataBroker view. This requires an LDB display addon, such as Titan Panel."],
          order=15,
          values= function()
            local feeds={}
            feeds[""]=L["None"]
            for name, func in pairs(Skada:GetFeeds()) do feeds[name]=name end
            return feeds
          end,
          get=function() return Skada.db.profile.feed end,
          set=function(self, feed)
            Skada.db.profile.feed=feed
            if feed~="" then Skada:SetFeed(Skada:GetFeeds()[feed]) end
          end
        },

        setnumber={
          type="toggle",
          name=L["Number set duplicates"],
          desc=L["Append a count to set names with duplicate mob names."],
          order=16,
          get=function() return Skada.db.profile.setnumber end,
          set=function() Skada.db.profile.setnumber=not Skada.db.profile.setnumber end,
        },

        setformat={
          type="select",
          name=L["Set format"],
          desc=L["Controls the way set names are displayed."],
          order=17,
          values= Skada:SetLabelFormats(),
          get=function() return Skada.db.profile.setformat end,
          set=function(self, opt) Skada.db.profile.setformat=opt; Skada:ApplySettings(); end,
          width="double",
        },

        setstokeep={
          type="range",
          name=L["Data segments to keep"],
          desc=L["The number of fight segments to keep. Persistent segments are not included in this."],
          order=98,
          min=0,
          max=30,
          step=1,
          get=function() return Skada.db.profile.setstokeep end,
          set=function(self, val) Skada.db.profile.setstokeep=val end,
        },

        updatefrequency={
          type="range",
          name=L["Update frequency"],
          desc=L["How often windows are updated. Shorter for faster updates. Increases CPU usage."],
          order=99,
          min=0.10,
          max=1,
          step=0.05,
          get=function() return Skada.db.profile.updatefrequency end,
          set=function(self, val) Skada.db.profile.updatefrequency=val end,
          width="double",
        }
      }
    },

    columns={
      type="group",
      name=L["Columns"],
      order=4,
      args={}
    },

    tooltips={
      type="group",
      name=L["Tooltips"],
      order=4,
      args={
        tooltips={
          type="toggle",
          name=L["Show tooltips"],
          desc=L["Shows tooltips with extra information in some modes."],
          order=1,
          get=function() return Skada.db.profile.tooltips end,
          set=function() Skada.db.profile.tooltips=not Skada.db.profile.tooltips end,
        },

        informative={
          type="toggle",
          name=L["Informative tooltips"],
          desc=L["Shows subview summaries in the tooltips."],
          order=2,
          get=function() return Skada.db.profile.informativetooltips end,
          set=function() Skada.db.profile.informativetooltips=not Skada.db.profile.informativetooltips end,
        },

        rows={
          type="range",
          name=L["Subview rows"],
          desc=L["The number of rows from each subview to show when using informative tooltips."],
          min=1,
          max=10,
          step=1,
          get=function() return Skada.db.profile.tooltiprows end,
          set=function(self, val) Skada.db.profile.tooltiprows=val end,
          order=3,
        },

        tooltippos={
          type="select",
          name=L["Tooltip position"],
          desc=L["Position of the tooltips."],
          values= {["default"]=L["Default"], ["topright"]=L["Top right"], ["topleft"]=L["Top left"], ["smart"]=L["Smart"]},
          get=function() return Skada.db.profile.tooltippos end,
          set=function(self, opt) Skada.db.profile.tooltippos=opt end,
          order=4,
        },
      }
    },

    modules={
      type="group",
      name=L["Disabled Modules"],
      order=6,
      get=function(i) return Skada.db.profile.modulesBlocked[i[#i]] end,
      set=function(i, value) Skada.db.profile.modulesBlocked[i[#i]]=value; Skada.options.args.modules.args.apply.disabled=false end,
      args={
        desc={
          type="description",
          name=L["Tick the modules you want to disable."],
          width="full",
          order=0,
        },
        apply={
          type="execute",
          name=APPLY,
          width="full",
          func=ReloadUI,
          confirm=function()
            return L["This change requires a UI reload. Are you sure?"]
          end,
          disabled=true,
          order=99,
        },
      },
    }
  }
}
